Source/DafnyLanguageServer/CounterExampleGeneration/DafnyModel.cs

```diff
@@ -52,47 +47,45 @@ public class DafnyModel {
     private static readonly Regex BvTypeRegex = new("^bv[0-9]+Type$");
     private static readonly Regex UnderscoreRemovalRegex = new("__");
 
-
     public DafnyModel(Model model, DafnyOptions options) {
       Model = model;
       this.options = options;
-      fSetSelect = MergeMapSelectFunctions(2);
-      fSeqLength = model.MkFunc("Seq#Length", 1);
-      fSeqBuild = model.MkFunc("Seq#Build", 2);
-      fSeqAppend = model.MkFunc("Seq#Append", 2);
-      fSeqDrop = model.MkFunc("Seq#Drop", 2);
-      fSeqTake = model.MkFunc("Seq#Take", 2);
-      fSeqIndex = model.MkFunc("Seq#Index", 2);
-      fSeqUpdate = model.MkFunc("Seq#Update", 3);
-      fSeqCreate = model.MkFunc("Seq#Create", 4);
-      fSeqEmpty = model.MkFunc("Seq#Empty", 1);
-      fSetEmpty = model.MkFunc("Set#Empty", 1);
-      fSetUnion = model.MkFunc("Set#Union", 2);
-      fSetUnionOne = model.MkFunc("Set#UnionOne", 2);
-      fSetIntersection = model.MkFunc("Set#Intersection", 2);
-      fSetDifference = model.MkFunc("Set#Difference", 2);
-      fMapDomain = model.MkFunc("Map#Domain", 1);
-      fMapElements = model.MkFunc("Map#Elements", 1);
-      fMapBuild = model.MkFunc("Map#Build", 3);
-      fIs = model.MkFunc("$Is", 2);
-      fIsBox = model.MkFunc("$IsBox", 2);
-      fBox = model.MkFunc("$Box", 1);
-      fDim = model.MkFunc("FDim", 1);
-      fIndexField = model.MkFunc("IndexField", 1);
-      fMultiIndexField = model.MkFunc("MultiIndexField", 2);
-      fDtype = model.MkFunc("dtype", 1);
-      fNull = model.MkFunc("null", 0);
-      fCharToInt = model.MkFunc("char#ToInt", 1);
-      fType = model.MkFunc("type", 1);
-      fChar = model.MkFunc("charType", 0);
-      fReal = model.MkFunc("realType", 0);
-      fU2Real = model.MkFunc("U_2_real", 1);
-      fBool = model.MkFunc("boolType", 0);
-      fU2Bool = model.MkFunc("U_2_bool", 1);
-      fInt = model.MkFunc("intType", 0);
-      fU2Int = model.MkFunc("U_2_int", 1);
-      fTag = model.MkFunc("Tag", 1);
-      fBv = model.MkFunc("TBitvector", 1);
+      var tyArgs = options.TypeEncodingMethod switch {
```

Comment:
```suggestion
      var tyArgMultiplier = options.TypeEncodingMethod switch {
```
I came up with the original name for this, but I think I didn't choose a good one. Maybe this is better?